packages/medusa/src/strategies/cart-completion.ts

```diff
@@ -326,6 +156,182 @@ class CartCompletionStrategy extends AbstractCartCompletionStrategy {
       response_code: idempotencyKey.response_code,
     }
   }
+
+  protected async handleStarted(
+    id: string,
+    { manager }: { manager: EntityManager }
+  ) {
+    const cart = await this.cartService_.withTransaction(manager).retrieve(id, {
+      relations: [
+        "customer",
+        "discounts",
+        "discounts.rule",
+        "gift_cards",
+        "items",
+        "items.adjustments",
+        "region",
+        "region.tax_rates",
+        "shipping_address",
+        "shipping_methods",
+      ],
+    })
+
+    if (cart.completed_at) {
+      return {
+        response_code: 409,
+        response_body: {
+          code: MedusaError.Codes.CART_INCOMPATIBLE_STATE,
+          message: "Cart has already been completed",
+          type: MedusaError.Types.NOT_ALLOWED,
+        },
+      }
+    }
+
+    await this.cartService_.withTransaction(manager).createTaxLines(cart)
+
+    return {
+      recovery_point: "tax_lines_created",
+    }
+  }
+
+  protected async handleTaxLineCreated(
+    id: string,
+    idempotencyKey: IdempotencyKey,
+    { context, manager }: { context: any; manager: EntityManager }
+  ) {
+    const cart = await this.cartService_
+      .withTransaction(manager)
+      .authorizePayment(id, {
+        ...context,
+        cart_id: id,
+        idempotency_key: idempotencyKey,
+        useExistingTaxLine: true,
```

Comment:
question: I don't think we are using this option any longer. Can you confirm or deny?

Reply:
we don't use it anymore I ve simplified